### PR TITLE
Add prerequisites for practice exercises

### PR DESCRIPTION
**Pull request by TomPradat**
This aims to add prerequisites for some exercises in https://github.com/exercism/javascript/issues/984

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -927,7 +927,7 @@
         "name": "Transpose",
         "uuid": "9c140fb7-cc8b-411b-b613-a0e0081a9c3f",
         "practices": [],
-        "prerequisites": [],
+        "prerequisites": ["arrays", "for-loops", "strings", "conditionals"],
```
Comment (TomPradat) [inline config.json:930]: Here is my solution : https://exercism.org/tracks/javascript/exercises/transpose/solutions/TomPradat

I'm kinda facing a dilemma here : Either we consider that we can solve this exercise like I did, i.e without `array-transformations` and/or `array-analysis` or we add the two latest prerequisites (as used in the majority of community solutions I've seen) but it's a shame given the difficulty of the exercise.

Either ways, I'd recommend increasing the difficulty to 3 or 4. This exercise resembles matrix whose difficulty is 3.

File: config.json [hunk 2]
```diff
@@ -910,7 +910,13 @@
...
         "uuid": "1f84305d-ea76-4fe2-9858-3b53576d683d",
         "practices": [],
-        "prerequisites": [],
+        "prerequisites": [
+          "strings",
+          "numbers",
+          "errors",
+          "conditionals",
+          "array-transformations"
+        ],
```
Comment (TomPradat) [inline config.json:919]: Here is my solution : https://exercism.org/tracks/javascript/exercises/largest-series-product/solutions/TomPradat

File: config.json [hunk 3]
```diff
@@ -885,7 +885,12 @@
         "name": "Scale Generator",
         "uuid": "b9c586e8-998b-4f5d-ab98-a08be29a9f19",
         "practices": [],
-        "prerequisites": [],
+        "prerequisites": [
+          "classes",
+          "arrays",
+          "strings",
+          "array-transformations"
+        ],
```
Comment (TomPradat) [inline config.json:893]: My solution : https://exercism.org/tracks/javascript/exercises/scale-generator/solutions/TomPradat 

I found [mstange22's](https://exercism.org/tracks/javascript/exercises/scale-generator/solutions/mstange22) and the [proof solution](https://github.com/exercism/javascript/blob/main/exercises/practice/scale-generator/.meta/proof.ci.js) more concise and understandable so I filled the prerequisites according to these

File: config.json (hunk 1 above)
Comment (SleeplessByte) [reply config.json:941]: Lets increase the difficulty and lets NOT add the array prereqs. If it's harder, people are bound to have more experience, so we'll see more people solving it your way.

3 is fine.

**Review by SleeplessByte — APPROVED** [review]
@TomPradat can you also increase the difficulty? Feel free to merge if you can, or ping me and I will.

File: config.json (hunk 1 above)
Comment (TomPradat) [reply config.json:941]: [118baee](https://github.com/exercism/javascript/pull/1974/commits/118baee78de925ffc25fc0c349ccec8898368dce)

**Comment by TomPradat** [comment]
> @TomPradat can you also increase the difficulty? Feel free to merge if you can, or ping me and I will.

Done @SleeplessByte ! I cannot merge on my own though

**Pull request merged**